### ISSUE TITLE
More options to configure instead of env vars

### ISSUE THREAD

**Issue by mnlevy1981** [ESMCI/cime#1265]
Several of the mapping tools rely on `tools/configure` prior to building, which provides the following help message

```
$ ./configure --help
usage: configure [-h] [-d] [-v] [-s] [--machine MACHINE]
                 [--machines-dir MACHINES_DIR]
                 [--macros-format {Makefile,CMake}] [--output-dir OUTPUT_DIR]

This script writes CIME build information to a directory. The pieces of
information that will be written include: 1. Machine-specific build settings
(i.e. the "Macros" file). 2. File-specific build settings (i.e. "Depends"
files). 3. Environment variable loads (i.e. the env_mach_specific files). The
.env_mach_specific.sh and .env_mach_specific.csh files are specific to a given
compiler, MPI library, and DEBUG setting. By default, these will be the
machine's default compiler, the machine's default MPI library, and FALSE,
respectively. These can be changed by setting the environment variables
COMPILER, MPILIB, and DEBUG, respectively.
```

It would be great if `COMPILER`, `MPILIB`, and `DEBUG` were additional command line options instead of environment-dependent... i.e.

```
$ ./configure --macros-format Makefile --mpilib mpi-serial
```

instead of

```
$ export MPILIB=mpi-serial ; ./configure --macros-format Makefile --mpilib mpi-serial
```

Also, a `--clean` option that removes the `.env.*` files (and everything else generated by `configure`) would be nice.

**Comment by mnlevy1981** [comment]
Also, there is a `Readme.configure` file in `tools/` that is out of date:

```
$ cat Readme.configure 
SYNOPSIS
     configure [options]
OPTIONS
     User supplied values are denoted in angle brackets (<>).  Any value that contains
     white-space must be quoted.  Long option names may be supplied with either single
     or double leading dashes.  A consequence of this is that single letter options may
     NOT be bundled.

     -cimeroot            Specify the toplevel cime directory.
                          default: use CIMEROOT environment variable
     -mach <name>         Specify a machine (required).
     -compiler <name>     Specify a compiler for the target machine (optional)
                          default: default compiler for the target machine
     -mpilib <name>       Specify an mpi library for the target machine (optional)
                          default: mpi-serial  (most of the tools are serial apps)
     -mach_dir <path>     Specify the locations of the Machines directory (optional).

     -output_dir <path>   default: current working directory
     -output_format <value> Output format can be make or cmake.
                          default: make
     -help [or -h]        Print usage to STDOUT (optional).
     -loglevel

EXAMPLES

  ./configure -mach bluefire -compiler ibm -cimeroot /path/to/cime


This tool is used to create Macros files to build CESM tools on supported machines.
It will create a Macros file in make or cmake format along with the module support
for the specified or default compiler.   It will also create two files
.env_mach_specific.csh
.env_mach_specific.sh

Before you try to run make you should source the file above appropriate for your shell
this will set the proper environment for supported systems which use modules.```

**Comment by mnlevy1981** [comment]
This was fixed with #1268
